frontend/src/composables/use-search.ts

```diff
@@ -6,14 +6,13 @@ import { useMediaStore } from "~/stores/media"
 
 import { useI18nResultsCount } from "~/composables/use-i18n-utilities"
 import { useMatchSearchRoutes } from "~/composables/use-match-routes"
-import type { EventName, Events } from "~/types/analytics"
+import { useAnalytics } from "~/composables/use-analytics"
 
-export const useSearch = (
-  sendCustomEvent: <T extends EventName>(name: T, payload: Events[T]) => void
-) => {
+export const useSearch = () => {
   const mediaStore = useMediaStore()
   const searchStore = useSearchStore()
   const router = useRouter()
+  const { sendCustomEvent } = useAnalytics()
```

Comment:
Have you checked that this always works? 

The reason that `sendCustomEvent` was passed as a parameter here was that you cannot use a composable that requires Nuxt instance outside of setup. There is an edge case `useSearch` is called, if I remember correctly, in `asyncData`, and Nuxt instance is not available, so a call to `useAnalytics` inside `useSearch` produces a warning.

Reply:
I see. But then should `useSearch` return a product based on calling `useAnalytics`? I'll revert this change, but that to me sounds like a bit of code smell, if the composable isn't able to call other composables at the top level. Maybe I am misunderstanding the edge case. Either way, I'll remove this from the PR, it definitely doesn't need to be here.